File: rb/features.rb
# KokkaiKaigirokuApi SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module KokkaiKaigirokuApiFeatures
  def self.make_feature(name)
    case name
    when "base"
      KokkaiKaigirokuApiBaseFeature.new
    when "ratelimit"
      KokkaiKaigirokuApiRatelimitFeature.new
    when "retry"
      KokkaiKaigirokuApiRetryFeature.new
    when "test"
      KokkaiKaigirokuApiTestFeature.new
    when "timeout"
      KokkaiKaigirokuApiTimeoutFeature.new
    else
      KokkaiKaigirokuApiBaseFeature.new
    end
  end
end

File: py/features.py
# KokkaiKaigirokuApi SDK feature factory

from feature.base_feature import KokkaiKaigirokuApiBaseFeature
from feature.test_feature import KokkaiKaigirokuApiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KokkaiKaigirokuApiBaseFeature(),
        "test": lambda: KokkaiKaigirokuApiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
